onnx/shape_inference/implementation.cc

```diff
@@ -684,6 +688,40 @@ void InferShapes(
   }
 }
 
+std::unordered_map<std::string, TensorShapeProto> InferShapesAndDataPropagation(
+    ModelProto& m,
+    const std::unordered_map<std::string, TensorShapeProto>& generated_shape_data_by_name,
```

Comment:
So, is this used because the caller iteratively uses results from previous calls to do inference for other subgraphs later?